public/hacks/field_mapper_hack.js

```diff
@@ -1,98 +1,101 @@
-                const defaultConfig = '{\n  \"index_pattern\": {\n    \"*\": {\n' +
-                      '\"include\": [],\n      \"exclude\": [\".*\"]\n    }\n  }\n}';
-                let settings = config.get('fieldMapperHack:fields', defaultConfig);
-                settings = settings.index_pattern;
-
-                let match = { includes: [], excludes: [] };
-
-                if (_.has(settings, indexPattern.id)) {
-                    match = settings[indexPattern.id];
-                }
-                else if (_.has(settings, '*')) {
-                    match = settings['*'];
-                }
-
-                const included = [];
-                const excluded = [];
-
-                _.forEach(match.include, function (expression) {
-                    _.forEach(paths, function (path) {
-                        if (path.match(expression)) {
-                            included.push(path);
-                        }
-                    });
-                });
-
-                _.forEach(match.exclude, function (expression) {
-                    _.forEach(included, function (path) {
-                        if (path.match(expression)) {
-                            excluded.push(path);
-                        }
-                    });
-                });
-
-                // 3) Add a field mapping for the missing parents
-                _.forEach(_.difference(included, excluded), function (path) {
-                    fields.push({
-                        name: path,
-                        aggregatable: false,
-                        searchable: true,
-                        analyzed: false,
-                        doc_values: false,
-                        indexed: true,
-                        type: 'string'
-                    });
-                });
-
-                return fields;
+              let paths = [];
+              const mappingNames = [];
+
+              // 1) Iterate the field names, identify the "parent" paths
+              _.forEach(fields, function (field) {
+
+                  const fieldName = field.name;
+                  mappingNames.push(fieldName);
+
+                  const parts = fieldName.split('.');
+
+                  while (parts.length > 1) {
+                      parts.pop();
+                      paths.push(parts.join('.'));
+                  }
+              });
+
+              paths = _.uniq(_.difference(paths, mappingNames));
+
+              // 2) Test the discovered field names against the configuration
+              const defaultConfig = '{\n  \"index_pattern\": {\n    \"*\": {\n' +
+                    '\"include\": [],\n      \"exclude\": [\".*\"]\n    }\n  }\n}';
+              let { index_pattern: settings } = config.get('fieldMapperHack:fields', defaultConfig);
+              let match = { includes: [], excludes: [] };
+
+              if (_.has(settings, indexPattern.id)) {
```

Comment:
This doesn't appear to be matching. The following configuration should match a single index pattern (my `pulse-*` index pattern) but it doesn't.
```
{
  "index_pattern": {
    "a9b80760-8768-11ea-96be-5500e2ace1d6": {
      "include": ["^meta.\\w"],
      "exclude": []
    }
  }
}
```

Reply:
Issue Fixed